data/random-teams.ts

```diff
@@ -2448,6 +2448,293 @@ export class RandomTeams {
 
 		return team;
 	}
+
+	randomBSSFactorySets: AnyObject = require("./bss-factory-sets.json");
+
+	randomBSSFactorySet(
+		species: Species,
+		teamData: RandomTeamsTypes.FactoryTeamDetails
+	): RandomTeamsTypes.RandomFactorySet | null {
+		const id = toID(species.name);
+		const setList = this.randomBSSFactorySets[id].sets;
+
+		const movesMax: { [k: string]: number } = {
+			batonpass: 1,
+			stealthrock: 1,
+			toxicspikes: 1,
+			trickroom: 1,
+			auroraveil: 1,
+		};
+		const weatherAbilities = [
+			"drizzle",
+			"drought",
+			"snowwarning",
+			"sandstream",
+		];
+		const terrainAbilities: { [k: string]: string } = {
+			electricsurge: "electric",
+			psychicsurge: "psychic",
+			grassysurge: "grassy",
+			seedsower: "grassy",
+			mistysurge: "misty",
+		};
+		const terrainItemsRequire: { [k: string]: string } = {
+			electricseed: "electric",
+			psychicseed: "psychic",
+			grassyseed: "grassy",
+			mistyseed: "misty",
+		};
+
+		const maxWantsTera = 2;
+
+		const requiredMoves: { [k: string]: number } = {};
+
+		// Build a pool of eligible sets, given the team partners
+		// Also keep track of sets with moves the team requires
+		let effectivePool: {set: AnyObject, moveVariants?: number[], itemVariants?: number, abilityVariants?: number}[] = [];
+		const priorityPool = [];
+		for (const curSet of setList) {
+			let reject = false;
+			let hasRequiredMove = false;
+
+			// limit to 2 dedicated tera users per team
+			if (curSet.wantsTera &&	teamData.wantsTeraCount && teamData.wantsTeraCount >= maxWantsTera)	continue;
```

Comment:
You need to update `teamData.wantsTeraCount` when updating team data in `randomBSSFactoryTeam()`, otherwise this check won't do anything.